### PR TITLE
Remove AWSCore dependency

### PR DESCRIPTION
**Pull request by mattBrzezinski**
## Overview
This pull request is targeting `MB/json-service` (which is built on-top of `MB/query-service`). I am targeting that branch to only show the differences between the two, when those two branches go in we need to update this target to `v1`.

TODO:
- [x] Merge #127 
- [x] Merge #128 
- [x] Target `v1`

This pull request removes usages of `AWSCore.jl` from our update. It updates the quantifications to properly use `AWS.AWSConfig` and `AWS.AWSCredentials`.

It also includes an extra return statement in `AWS.do_request()` to handle the response type of `text/html`.

### CODE REVIEW

File: test/AWSCredentials.jl [hunk 1]
```diff
@@ -377,7 +365,7 @@ end
         "Security-Credentials" => "Test-Security-Credentials"
     )
 
-    http_get_patch = @patch function http_get(url::String)
+    _http_get_patch = @patch function HTTP.request(method::String, url::String)
```
Comment (iamed2) [inline test/AWSCredentials.jl:368]: Maybe rename to `_http_request_patch`?

File: test/AWS.jl [hunk 2]
```diff
@@ -371,10 +371,46 @@ end
...
+            @test_throws AWSException _get_secret_string(secret_name)
+        end
+    end
+end
+
 @testset "query" begin
     @testset "iam" begin
         policy_arn = ""
-        expected_policy_name = "AWS.jl-Test-Policy"
+        expected_policy_name = "AWS.jl-Test-Policy" * lowercase(Dates.format(now(Dates.UTC), "yyyymmddTHHMMSSZ"))
```
Comment (iamed2) [inline test/AWS.jl:413]: I'd feel more comfortable if you included some milliseconds on these.

Comment (mattBrzezinski) [reply test/AWS.jl:413]: Ugh.. this rebasing on various branches is going to make reviewing all these PRs a bit messy. :/

Comment (mattBrzezinski) [reply test/AWS.jl:413]:
```
MB/query-service -> MB/json-service -> MB/remove-aws-core
```

When changes are made down in `query-service` rebasing them throughout is a bit of a PITA.

File: test/AWSCredentials.jl (hunk 1 above)
Comment (mattBrzezinski) [reply test/AWSCredentials.jl:368]: Good catch, I'll update this.

File: test/AWS.jl [hunk 3]
```diff
@@ -381,7 +415,7 @@ end
 
 @testset "json" begin
     @testset "secrets_manager" begin
-        secret_name = "AWS.jl-Test-Secret" * lowercase(Dates.format(now(Dates.UTC), "yyyymmddTHHMMSSZ"))
+        secret_name = "AWS.jl-Test-Secret" * lowercase(Dates.format(now(Dates.UTC), "yyyymmddTHHMMSSsssZ"))
```
Comment (omus) [inline test/AWS.jl:418]: Just a best practise:
```suggestion
        secret_name = "AWS.jl-Test-Secret" * lowercase(Dates.format(now(Dates.UTC), dateformat"yyyymmdd\THHMMSSsssZ"))
```

File: test/AWS.jl [hunk 4]
```diff
@@ -412,7 +446,7 @@ end
 @testset "query" begin
     @testset "iam" begin
         policy_arn = ""
-        expected_policy_name = "AWS.jl-Test-Policy" * lowercase(Dates.format(now(Dates.UTC), "yyyymmddTHHMMSSZ"))
+        expected_policy_name = "AWS.jl-Test-Policy" * lowercase(Dates.format(now(Dates.UTC), "yyyymmddTHHMMSSsssZ"))
```
Comment (omus) [inline test/AWS.jl:449]: Consider making the date format a constant or making this into a function:
```suggestion
        expected_policy_name = "AWS.jl-Test-Policy" * lowercase(Dates.format(now(Dates.UTC), dateformat"yyyymmdd\THHMMSSsssZ"))
```

File: test/AWS.jl [hunk 5]
```diff
@@ -442,7 +476,7 @@ end
     end
 
     @testset "sqs" begin
-        queue_name = "aws-jl-test---" * lowercase(Dates.format(now(Dates.UTC), "yyyymmddTHHMMSSZ"))
+        queue_name = "aws-jl-test---" * lowercase(Dates.format(now(Dates.UTC), "yyyymmddTHHMMSSsssZ"))
```
Comment (omus) [inline test/AWS.jl:479]:
```suggestion
        queue_name = "aws-jl-test---" * lowercase(Dates.format(now(Dates.UTC), dateformat"yyyymmdd\THHMMSSsssZ"))
```

File: test/AWS.jl [hunk 6]
```diff
@@ -503,7 +537,7 @@ end
 end
 
 @testset "rest-xml" begin
-    bucket_name = "aws.jl-test---" * lowercase(Dates.format(now(Dates.UTC), "yyyymmddTHHMMSSZ"))
+    bucket_name = "aws.jl-test---" * lowercase(Dates.format(now(Dates.UTC), "yyyymmddTHHMMSSsssZ"))
```
Comment (omus) [inline test/AWS.jl:540]:
```suggestion
    bucket_name = "aws.jl-test---" * lowercase(Dates.format(now(Dates.UTC), dateformat"yyyymmdd\THHMMSSsssZ"))
```

File: test/AWS.jl [hunk 7]
```diff
@@ -580,7 +614,7 @@ end
 end
 
 @testset "rest-json" begin
-    timestamp = lowercase(Dates.format(now(Dates.UTC), "yyyymmddTHHMMSSZ"))
+    timestamp = lowercase(Dates.format(now(Dates.UTC), "yyyymmddTHHMMSSsssZ"))
```
Comment (omus) [inline test/AWS.jl:617]:
```suggestion
    timestamp = lowercase(Dates.format(now(Dates.UTC), dateformat"yyyymmdd\THHMMSSsssZ"))
```

File: test/AWSCredentials.jl [hunk 8]
```diff
@@ -14,77 +14,76 @@ macro test_ecode(error_codes, expr)
...
-        :resource => "/invenia-static-website-content/invenia_ca/index.html",
-        :url => "https://s3.us-east-1.amazonaws.com/invenia-static-website-content/invenia_ca/index.html",
-        :verb => "GET",
-        :region => "us-east-1",
-        :creds => nothing,
+    pub_request1 = AWS.Request(
+        service="s3",
+        api_version="2006-03-01",
+        request_method="GET",
+        headers=Dict{String, String}("Range"=>"bytes=0-0"),
```
Comment (omus) [inline test/AWSCredentials.jl:47]:
```suggestion
        headers=Dict("Range" => "bytes=0-0"),
```

File: test/AWSCredentials.jl [hunk 9]
```diff
@@ -14,77 +14,76 @@ macro test_ecode(error_codes, expr)
...
-        :url => "https://s3.amazonaws.com/ryft-public-sample-data/AWS-x86-AMI-queries.json",
-        :verb => "GET",
-        :region => "us-east-1",
-        :creds => nothing,
+
+    pub_request2 = AWS.Request(
+        service="s3",
+        api_version="2006-03-01",
+        request_method="GET",
+        headers=Dict{String, String}("Range"=>"bytes=0-0"),
```
Comment (omus) [inline test/AWSCredentials.jl:57]:
```suggestion
        headers=Dict("Range" => "bytes=0-0"),
```

File: test/AWSCredentials.jl [hunk 10]
```diff
@@ -361,7 +349,7 @@ end
         "Test-Profile" => "test",
         "Test-Config-Profile" => "profile test",
 
-        # Default profile values, needs to match due to AWSCredentials.jl:239
+        # Default profile values, needs to match due to AWSCredentials.jl check_credentials()
```
Comment (omus) [inline test/AWSCredentials.jl:352]: Mistake?

File: test/AWSCredentials.jl [hunk 11]
```diff
@@ -488,39 +472,34 @@ end
...
             @test aws_creds.access_key_id == test_values["AccessKeyId"]
             @test aws_creds.secret_key == test_values["SecretAccessKey"]
         end
     end
 
     @testset "Instance - EC2" begin
-        apply([http_get_patch]) do
-            # TODO: Remove qualification when dep on AWSCore is removed
-            result = AWS.ec2_instance_credentials()
+        apply([_http_request_patch]) do
```
Comment (omus) [inline test/AWSCredentials.jl:482]: You can also just pass in the patch directly:
```suggestion
        apply(_http_request_patch) do
```

File: test/AWSCredentials.jl [hunk 12]
```diff
@@ -488,39 +472,34 @@ end
...
+            @test result.renew == ec2_instance_credentials
         end
     end
 
     @testset "Instance - ECS" begin
         withenv("AWS_CONTAINER_CREDENTIALS_RELATIVE_URI" => test_values["URI"]) do
-            apply([http_get_patch]) do
-                # TODO: Remove qualification when dep on AWSCore is removed
-                result = AWS.ecs_instance_credentials()
+            apply([_http_request_patch]) do
```
Comment (omus) [inline test/AWSCredentials.jl:495]:
```suggestion
            apply(_http_request_patch) do
```

File: test/AWSCredentials.jl (hunk 10 above)
Comment (mattBrzezinski) [reply test/AWSCredentials.jl:352]: I remove `239` as this line number will not be accurate as time goes on, but the function name should continue.

File: test/AWS.jl [hunk 13]
```diff
@@ -381,7 +415,7 @@ end
 
 @testset "json" begin
     @testset "secrets_manager" begin
-        secret_name = "AWS.jl-Test-Secret" * lowercase(Dates.format(now(Dates.UTC), "yyyymmddTHHMMSSZ"))
+        secret_name = "AWS.jl-Test-Secret" * lowercase(Dates.format(now(Dates.UTC), dateformat"yyyymmdd\THHMMSSsssZ"))
```
Comment (omus) [inline test/AWS.jl:418]: Maybe you missed my [original comment](https://github.com/JuliaCloud/AWS.jl/pull/130#discussion_r459067981) but maybe you want to turn:
```julia
lowercase(Dates.format(now(Dates.UTC), dateformat"yyyymmdd\THHMMSSsssZ")
```
into a function?

File: test/AWSCredentials.jl [hunk 14]
```diff
@@ -14,77 +14,76 @@ macro test_ecode(error_codes, expr)
...
     @test_ecode(
         "InvalidAction",
-        AWSCore.Services.iam("GetFoo", Dict("ContentType" => "JSON"))
+        AWSServices.iam("GetFoo")
     )
 
     @test_ecode(
         ["AccessDenied", "NoSuchEntity"],
-        AWSCore.Services.iam("GetUser", Dict("UserName" => "notauser", "ContentType" => "JSON"))
+        AWSServices.iam("GetUser", Dict("UserName"=>"notauser"))
```
Comment (omus) [inline test/AWSCredentials.jl:28]:
```suggestion
        AWSServices.iam("GetUser", Dict("UserName" => "notauser"))
```

File: test/AWSCredentials.jl [hunk 15]
```diff
@@ -14,77 +14,76 @@ macro test_ecode(error_codes, expr)
...
     )
 
     @test_ecode(
         "ValidationError",
-        AWSCore.Services.iam("GetUser", Dict("UserName" => "@#!%%!", "ContentType" => "JSON"))
+        AWSServices.iam("GetUser", Dict("UserName"=>"@#!%%!"))
```
Comment (omus) [inline test/AWSCredentials.jl:33]:
```suggestion
        AWSServices.iam("GetUser", Dict("UserName" => "@#!%%!"))
```

File: test/AWSCredentials.jl [hunk 16]
```diff
@@ -14,77 +14,76 @@ macro test_ecode(error_codes, expr)
...
     )
 
     @test_ecode(
         ["AccessDenied", "EntityAlreadyExists"],
-        AWSCore.Services.iam("CreateUser", Dict("UserName" => "root", "ContentType" => "JSON"))
+        AWSServices.iam("CreateUser", Dict("UserName"=>"root"))
```
Comment (omus) [inline test/AWSCredentials.jl:38]:
```suggestion
        AWSServices.iam("CreateUser", Dict("UserName" => "root"))
```

File: test/AWS.jl (hunk 13 above)
Comment (mattBrzezinski) [reply test/AWS.jl:418]: Oh whoops, yes definitely missed that. Good idea!

File: src/AWSCredentials.jl [hunk 17]
```diff
@@ -492,13 +487,12 @@ function _aws_get_role(role::AbstractString, ini::Inifile)
...
     credentials === nothing && return nothing
-    config = AWSConfig(:creds => credentials, :region => aws_get_region(source_profile, ini))
+    config = AWSConfig(creds=credentials, region=aws_get_region(source_profile, ini))
 
     role = AWSServices.sts(
         config,
         "AssumeRole",
-        RoleArn=role_arn,
-        RoleSessionName=replace(role, r"[^\w+=,.@-]" => s"-"),
+        LittleDict("RoleArn"=>role_arn, "RoleSessionName"=>replace(role, r"[^\w+=,.@-]" => s"-"))
```
Comment (omus) [inline src/AWSCredentials.jl:495]:
```suggestion
        LittleDict("RoleArn" => role_arn, "RoleSessionName" => replace(role, r"[^\w+=,.@-]" => s"-"))
```

File: test/AWS.jl [hunk 18]
```diff
@@ -1,3 +1,7 @@
+function _get_formmated_date_now()
```
Comment (omus) [inline test/AWS.jl:1]: Typo
```suggestion
function _get_formatted_date_now()
```
Alternatively:
```suggestion
function _now_formatted()
```

**Pull request merged**